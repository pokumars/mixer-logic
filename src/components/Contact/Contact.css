.contact{
  display:flex;
  flex-direction: column;
  margin: 20px auto;
  padding: 5px;
  max-width: 400px;
  /* background-color: #EDF5E1; */
 background-color: #7e7b7b;
}
.my-name{
  margin: 15px auto;
}
.socials{
  display:flex;
  flex-direction: row;
  justify-content: center;
}
.socials img, .my-cv img, .my-email img{
  height: 50px;
}
.socials a{
  display: flex;
  flex-direction: column;
  padding:10px;
  margin: 10px;
  border-radius: 10px;
  /* color: #05386B; */
  color: #f09b9b;
  /* border: 1px solid #05386B; */
  border: 1px solid #f09b9b;
}

.my-email{
  display: flex;
}

.my-cv{
display: flex;
}