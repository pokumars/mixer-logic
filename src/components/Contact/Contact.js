import React from 'react'
import './Contact.css'

const Contact = () => {
  return (
    <div className="contact">
      <h2 className="my-name">Oheneba Poku-Marboah</h2>
      <div className="socials">
        <a href="https://www.linkedin.com/in/oheneba-poku-marboah-483783173/" className="contact-element">
          <img alt="linkedin" src={require('../../graphics/icons/linkedin-icon.svg')}/>
          <strong>LinkedIn</strong>
        </a>
        <a href="https://github.com/pokumars" className="contact-element">
          <img alt="github" src={require('../../graphics/icons/github-icon.svg')}/>
          <strong>GitHub</strong>
        </a>
      </div>
      <div className="my-email contact-element">
        <img alt="cv" src={require('../../graphics/icons/email-icon.svg')}/>
        <p>Shoot me some feedback @ <br/> ohenebakob@yahoo.com</p>
      </div>
      <div className="my-cv contact-element">
        <img alt="cv" src={require('../../graphics/icons/cv-icon.svg')}/>
        <p><a href={require('../../graphics/cv_oheneba_pokumarboah.pdf')} target="_blank" rel="noopener nonreferrer">
          my curriculum vitae
        </a></p>
      </div>
      
    </div>
  )
}

export default Contact
