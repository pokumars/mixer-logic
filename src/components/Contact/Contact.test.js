//This component barely needs any tests. i am only testing it for rehearsal sake
import React from 'react'
import {render, cleanup} from '@testing-library/react'
import Contact from './Contact'
const expectedLinkedIn = "https://www.linkedin.com/in/oheneba-poku-marboah-483783173/"
const expectedGithub = "https://github.com/pokumars"
const expectedCVLink = "cv_oheneba_pokumarboah.pdf"

afterEach(cleanup)
 
it('should take a snapshot', () => {
    const { asFragment } = render(<Contact />)
    
    expect(asFragment(<Contact />)).toMatchSnapshot()
});
describe('main Contact.js tests', () => {
    let contactComponent
    beforeEach(() => {
        contactComponent = render (<Contact />)
    })
    
    test('should have correct github href', () => {
        const githubLink = contactComponent.getByText('GitHub')

        expect(githubLink.closest('a')).toHaveAttribute('href', expectedGithub)
        expect(githubLink.closest('a')).toBeDefined()
    })

    test('should have correct LinkedIn href', () => {
        const linkedinLink = contactComponent.getByText('LinkedIn')

        expect(linkedinLink.closest('a')).toBeDefined()
        expect(linkedinLink.closest('a')).toHaveAttribute('href', expectedLinkedIn)
    })

    test('should have my cv that opens in a new tab', () => {
        const cvLink = contactComponent.getByText('curriculum vitae')

        expect(cvLink.closest('a')).toBeDefined()
        expect(cvLink.closest('a')).toHaveAttribute('href', expectedCVLink)
        expect(cvLink.closest('a')).toHaveAttribute('target', '_blank')
    })
})

