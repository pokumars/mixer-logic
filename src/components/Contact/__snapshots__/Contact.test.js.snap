// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="contact"
  >
    <h2
      class="my-name"
    >
      Oheneba Poku-Marboah
    </h2>
    <div
      class="socials"
    >
      <a
        class="contact-element"
        href="https://www.linkedin.com/in/oheneba-poku-marboah-483783173/"
      >
        <img
          alt="linkedin"
          src="[object Object]"
        />
        <strong>
          LinkedIn
        </strong>
      </a>
      <a
        class="contact-element"
        href="https://github.com/pokumars"
      >
        <img
          alt="github"
          src="[object Object]"
        />
        <strong>
          GitHub
        </strong>
      </a>
    </div>
    <div
      class="my-email contact-element"
    >
      <img
        alt="cv"
        src="[object Object]"
      />
      <p>
        Shoot me some feedback @ ohenebakob@yahoo.com
      </p>
    </div>
    <div
      class="my-cv contact-element"
    >
      <img
        alt="cv"
        src="[object Object]"
      />
      <p>
        <a
          href="cv_oheneba_pokumarboah.pdf"
          rel="noopener noreferrer"
          target="_blank"
        >
          curriculum vitae
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
